sentry-android-navigation/src/test/java/io/sentry/android/navigation/SentryNavigationListenerTest.kt

```diff
@@ -0,0 +1,119 @@
+package io.sentry.android.navigation
+
+import androidx.core.os.bundleOf
+import androidx.navigation.NavController
+import androidx.navigation.NavDestination
+import androidx.test.ext.junit.runners.AndroidJUnit4
+import com.nhaarman.mockitokotlin2.check
+import com.nhaarman.mockitokotlin2.mock
+import com.nhaarman.mockitokotlin2.reset
+import com.nhaarman.mockitokotlin2.verify
+import com.nhaarman.mockitokotlin2.whenever
+import io.sentry.Breadcrumb
+import io.sentry.IHub
+import io.sentry.SentryLevel
+import org.junit.runner.RunWith
+import org.robolectric.annotation.Config
+import kotlin.test.Test
+import kotlin.test.assertEquals
+import kotlin.test.assertNull
+
+@RunWith(AndroidJUnit4::class)
+@Config(sdk = [31])
+class SentryNavigationListenerTest {
+
+    class Fixture {
+        val hub = mock<IHub>()
+        val destination = mock<NavDestination>()
+        val navController = mock<NavController>()
+
+        fun getSut(toRoute: String = "route"): SentryNavigationListener {
+            whenever(destination.route).thenReturn(toRoute)
+            return SentryNavigationListener(hub)
+        }
+    }
+
+    private val fixture = Fixture()
```

Comment:
`l`: The fixture is a global state for all tests in this class. We set a permanent state to the `destination.route` with `whenever(destination.route).thenReturn(toRoute)`, which is not a problem at the moment because you only can create a sut by modifying this state, which might turn into a problem in the future resulting in stateful flaky tests. Maybe it's better to create a new fixture for every test?

Reply:
Ah didn't know that. Maybe it was different in the past. Thanks for pointing that out.